File: Java/백준/Bronze/24638. Anno Domini 2022/Anno Domini 2022.java
import java.util.*;
import java.io.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;

    public static void main(String[] args) throws IOException {

        st = new StringTokenizer(br.readLine());
        String a, b, c, d;
        int x = 0, y = 0;
        a = st.nextToken();
        b = st.nextToken();

        if (a.equals("AD"))
            x = Integer.parseInt(b) - 1;
        else
            x = -Integer.parseInt(a);

        st = new StringTokenizer(br.readLine());
        c = st.nextToken();
        d = st.nextToken();
        if (c.equals("AD"))
            y = Integer.parseInt(d) - 1;
        else
            y = -Integer.parseInt(c);

        bw.write(Math.abs(x - y) + "\n");
        bw.flush();
        bw.close();
    }
}

File: Python/백준/Bronze/24638. Anno Domini 2022/Anno Domini 2022.py
from sys import stdin

input = stdin.readline

x, y = 0, 0
f = list(input().split())
s = list(input().split())

if f[0] == "AD":
    x = int(f[1]) - 1
else:
    x = -int(f[0])

if s[0] == "AD":
    y = int(s[1]) - 1
else:
    y = -int(s[0])

print(abs(x - y))
